### PR TITLE
C++11 cleanup on constructors

### PR DESCRIPTION
**Pull request by Dumbeldor**

### CODE REVIEW

File: src/script/cpp_api/s_base.cpp [hunk 1]
```diff
@@ -72,8 +72,7 @@ class ModNameStorer
 */
 
 ScriptApiBase::ScriptApiBase() :
-	m_luastackmutex(),
-	m_gamedef(NULL)
+	m_luastackmutex()
```
Comment (nerzhul) [inline src/script/cpp_api/s_base.cpp]: this constructor is useless you can remove it

**Comment by nerzhul** [comment]
@Zeno- @SmallJoker can you take a look ?

File: src/script/lua_api/l_areastore.h [hunk 2]
```diff
@@ -49,7 +49,7 @@ class LuaAreaStore : public ModApiBase
 	static int l_from_file(lua_State *L);
 
 public:
-	AreaStore *as;
+	AreaStore *m_as = nullptr;
```
Comment (SmallJoker) [inline src/script/lua_api/l_areastore.h]: `m_` prefix is used for `private` members. Revert to `as` or `areastore`.

File: src/threading/event.cpp [hunk 3]
```diff
@@ -32,7 +32,6 @@ void Event::wait()
 	while (!notified) {
 		cv.wait(lock);
 	}
-	notified = false;
```
Comment (nerzhul) [inline src/threading/event.cpp]: why this disappears ?

**Pull request merged**